src/parser_error.rs

```diff
@@ -0,0 +1,972 @@
+use std::{
+    convert::TryFrom,
+    net::AddrParseError,
+    num::{IntErrorKind, ParseFloatError, ParseIntError},
+    str::ParseBoolError,
+};
+
+use pyo3::prelude::*;
+use tantivy::{self as tv, schema::FacetParseError};
+
+// TODO(https://github.com/PyO3/pyo3/issues/1190): Expose this to bindings once trait <-> ABC is
+// supported in PyO3.
+pub(crate) trait QueryParserError {
```

Comment:
This isn't a base class because it seems base classes make it tough to do things like `IntoPy` and upcasting.

See: https://github.com/PyO3/pyo3/issues/1836 and https://github.com/PyO3/pyo3/issues/787